community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -900,12 +900,32 @@ public long nextCommittingTransactionId()
     @Override
     public void transactionCommitted( long transactionId, long checksum )
     {
-        if ( lastCommittedTx.offer( transactionId, checksum ) )
+        checkInitialized( lastCommittingTxField.get() );
+        TransactionId current = highestCommittedTransaction.get();
+        if ( current.transactionId() < transactionId )
```

Comment:
On that note, we could break out the contents of this method as a class `HighestCommittedTransactionId` so that we could test this in isolation... just like how we do with `OutOfOrderSequence`.